### PR TITLE
Update provider to 46a27996b973bbd48cabec587b9a419cd4319add

### PR DESCRIPTION
**Pull request by goderbauer**
Updates provider to the latest hash.

### CODE REVIEW

**Comment by goderbauer** [comment]
cc @rami-a Looks like provider still has code that isn't null safe? Unfortunately, upcoming Dart 3 will no longer support that as announced in https://medium.com/dartlang/the-road-to-dart-3-afdd580fbefa and our master channel has already been updated to enforce that restriction (see failing checks).

File: registry/provider.test [hunk 1]
```diff
@@ -1,6 +1,6 @@
 contact=darky12s@gmail.com
 fetch=git clone https://github.com/rrousselGit/provider.git tests
-fetch=git -C tests checkout faa2d4fd97a5d95345c95525250cb0ed2602003b
+fetch=git -C tests checkout 1b3374c282824f95b12b16ab22d6a718912b4a6b
 update=.
 test=flutter analyze --no-fatal-infos
 test=flutter test
```
Comment (rrousselGit) [inline registry/provider.test:6]:
```suggestion
fetch=git -C tests checkout 46a27996b973bbd48cabec587b9a419cd4319add
update=.
test=flutter analyze --no-fatal-infos
test=flutter test test/null_safe
```

This should work. I've moved mixed mode tests to a separate folder

**Comment by goderbauer** [comment]
@rrousselGit Looks like tests are passing now. Thanks for your work. Would you mind approving/LGTMing this PR? Thanks again!

**Review by rrousselGit — COMMENTED** [review]
Sure, LGTM :)

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**